lib/pages/settings/view/admin_page.dart

```diff
@@ -20,49 +22,71 @@ class _AdminPanelPageState extends State<AdminPanelPage> {
-                }
-              },
-            })
-      ],
-      title: Text(S.current.navigationAdmin),
-      body: FutureBuilder(
-          future: all
-              ? adminProvider.fetchAllRequestIds()
-              : adminProvider.fetchUnprocessedRequestIds(),
-          builder: (context, snapshot) {
-            if (snapshot.connectionState == ConnectionState.done) {
-              return ListView.builder(
-                itemCount:
-                    snapshot.data?.length != null ? snapshot.data.length : 0,
-                itemBuilder: (context, index) {
-                  return RequestCard(
-                    requestId: snapshot.data[index],
-                  );
+    if (checkUserIsAdmin()) {
+      final adminProvider = Provider.of<AdminProvider>(context, listen: false);
+      return AppScaffold(
+        actions: [
+          AppScaffoldAction(
+              icon: FeatherIcons.filter,
+              tooltip: S.current.navigationFilter,
+              items: {
+                S.current.filterMenuShowAll: () {
+                  if (!all) {
+                    setState(() => all = true);
+                  } else {
+                    AppToast.show(S.current.warningFilterAlreadyAll);
+                  }
                 },
-              );
-            } else {
-              return const Center(child: CircularProgressIndicator());
-            }
-          }),
-    );
+                S.current.filterMenuShowUnprocessed: () {
+                  if (all) {
+                    setState(() => all = false);
+                  } else {
+                    AppToast.show(S.current.warningFilterAlreadyUnprocessed);
+                  }
+                },
+              })
+        ],
+        title: Text(S.current.navigationAdmin),
+        body: FutureBuilder(
+            future: all
+                ? adminProvider.fetchAllRequestIds()
+                : adminProvider.fetchUnprocessedRequestIds(),
+            builder: (context, snapshot) {
+              if (snapshot.connectionState == ConnectionState.done) {
+                return ListView.builder(
+                  itemCount:
+                      snapshot.data?.length != null ? snapshot.data.length : 0,
+                  itemBuilder: (context, index) {
+                    return RequestCard(
+                      requestId: snapshot.data[index],
+                    );
+                  },
+                );
+              } else {
+                return const Center(child: CircularProgressIndicator());
+              }
+            }),
+      );
+    } else {
+      return AppScaffold(
+        body: ErrorPage(
+          imgPath: 'assets/illustrations/undraw_access_denied.png',
+          errorMessage: S.current.errorDontHavePermissions,
+        ),
+      );
+    }
+  }
+
+  bool checkUserIsAdmin() {
+    final authProvider = Provider.of<AuthProvider>(context, listen: false);
```

Comment:
I believe `listen: false` isn't necessary here. (also now that I look at it it's probably not necessary above either, for the admin provider)